platform/darwin/src/image.mm

```diff
@@ -81,6 +81,34 @@ CGContextHandle context(CGBitmapContextCreate(
     return image;
 }
 
+mbgl::UnassociatedImage MGLUnassociatedImageFromCGImage(CGImageRef src) {
+    const size_t width = CGImageGetWidth(src);
+    const size_t height = CGImageGetHeight(src);
+
+    mbgl::UnassociatedImage image({ static_cast<uint32_t>(width), static_cast<uint32_t>(height) });
+
+    CGColorSpaceHandle colorSpace(CGColorSpaceCreateDeviceRGB());
+    if (!colorSpace) {
+        throw std::runtime_error("CGColorSpaceCreateDeviceRGB failed");
+    }
+
+    constexpr const size_t bitsPerComponent = 8;
+    constexpr const size_t bytesPerPixel = 4;
+    const size_t bytesPerRow = bytesPerPixel * width;
+
+    CGContextHandle context(CGBitmapContextCreate(
+        image.data.get(), width, height, bitsPerComponent, bytesPerRow, *colorSpace,
+        kCGBitmapByteOrderDefault | kCGImageAlphaNoneSkipLast));
```

Comment:
It seems like `kCGImageAlphaNoneSkipLast` completely drops the alpha channel rather than decoding it in a non-premultiplied fashion. Did you check that it in fact /does/ have alpha?